### PR TITLE
docker auth: improve missing user/pwd

### PR DESCRIPTION
**Pull request by Frankkkkk**
Specify the "missing username and password" error message. This makes debugging the action easier when for example mistyping the username or the password.

Thanks and cheers!

### CODE REVIEW

**Comment by crazy-max** [comment]
Thanks for this PR but I think the error message to users should remain generic. You can still check what value is empty in the login step (expand `Run docker/login-action@v3`):

image

File: src/docker.ts [hunk 1]
```diff
@@ -21,9 +21,16 @@ export async function logout(registry: string): Promise<void> {
...
+  if (!username && !password) {
     throw new Error('Username and password required');
   }
+  if (!username) {
+    throw new Error('Username required');
+  }
+  if (!password) {
+    throw new Error('Password required');
+  }
+
```
Comment (crazy-max) [inline src/docker.ts:33]: remove this extra new line

**Review by crazy-max — CHANGES_REQUESTED** [review]
Ok this change sounds good considering how other providers currently consume GitHub Actions.

You forgot to run `docker buildx bake pre-checkin` to update generated content btw, see https://github.com/docker/login-action/blob/master/.github/CONTRIBUTING.md#submitting-a-pull-request

Don't forget to squash commits as well. Thanks.

**Comment by Frankkkkk** [comment]
Done, thanks.
Curious question: the dist file are so large; how do you make sure that a malicious user didn't embed some nefarious content in it ?
Subsidiary one: how do you deal with conflicts on MRs ? Do all contributors always have to rebase on every change on main?
Thanks!

**Review by crazy-max — APPROVED** [review]
LGTM thanks!

**Pull request merged**